src/node/rpc/frontend.h

```diff
@@ -284,7 +286,18 @@ namespace ccf
             return ctx->serialise_response();
           }
 
-          switch (tx.commit())
+          std::function<kv::Version()> version_resolver = nullptr;
+          if (reserved != -1)
+          {
+            CCF_ASSERT(
+              consensus->type() == ConsensusType::BFT, "Wrong consensus type");
+            version_resolver = [&]() {
```

Comment:
I'm a little worried about what this means for anything else trying to interact with the Store. If I'm reading correctly, the idea is that we call `next_version()` n times, but the `reserved` Version we _claim_ may happen out of order? But this seems like a complication to the `Store` API - if any other system is trying to read "where are you", it gets junk and needs to know that these are applied out-of-order. It would be simpler if we could execute our batch on the side (not touching the KV so other systems can safely call its old values), and then update it with the final state all at once? Obviously a large architectural change, but we need to think about what the semantics of any of these calls actually are.

Reply:
I do not agree with this statement. The only guarantee that we provide our users is around globally committed state in regards to stability and serializability. In regards to correctness in a BFT system everything that reads from the KV needs to go through the consensus otherwise the result you have is junk anyway.

Reply:
Sorry I don't mean to users, in the endpoint interface. I mean other bits of code within CCF. A lot of the instance I was thinking of are only in unit tests, so don't really matter.

But for instance, how does a snapshot work when `current_version()` doesn't actually report what's in the KV? We also ask for `current_version()` several times during recovery - how do we ensure that's not affected by this change? (There may be answers already for these! Just want to make sure they're considered and documented)

Reply:
This is done the same way that we deal with view-changes and signature transactions on the backup in this PR. We ensure that there in no pending execution when we trigger these cases.